saleor/checkout/complete_checkout.py

```diff
@@ -764,15 +763,29 @@ def complete_checkout(
     """
     if site_settings is None:
         site_settings = Site.objects.get_current().settings
+    checkout_for_update = (
+        Checkout.objects.select_for_update()
+        .filter(pk=checkout_info.checkout.pk)
+        .first()
+    )
 
     fetch_checkout_prices_if_expired(
-        checkout_info, manager, lines, discounts=discounts, site_settings=site_settings
+        checkout_info,
+        manager,
+        lines,
+        discounts=discounts,
+        site_settings=site_settings,
     )
 
-    checkout = checkout_info.checkout
-    channel_slug = checkout_info.channel.slug
-    payment = checkout.get_last_active_payment()
-    try:
+    with transaction_with_commit_on_errors():
+        checkout = checkout_info.checkout
+        if not checkout_for_update:
+            order = Order.objects.filter(checkout_token=checkout.token).first()
+            if order:
+                return order, False, {}
+
+        channel_slug = checkout_info.channel.slug
+        payment = checkout.get_last_active_payment()
```

Comment:
Shouldn't we use `checkout_for_update` instead of `checkout = checkout_info.checkout` inside this transaction? 
```suggestion
        payment = checkout_for_update.get_last_active_payment()
```